src/core/gatherer.ts

```diff
@@ -73,18 +74,23 @@ export class Gatherer {
     return userAgent + syntheticsIdentifier;
   }
 
-  static async setNetworkConditions(
+  static setNetworkConditions(
     context: BrowserContext,
     networkConditions: RunOptions['networkConditions']
   ) {
     if (networkConditions) {
-      context.on('page', async page => {
+      context.on('page', page => {
         const context = page.context();
-        const client = await context.newCDPSession(page);
-        await client.send(
-          'Network.emulateNetworkConditions',
-          networkConditions
-        );
+        const emulatePromise = context
+          .newCDPSession(page)
+          .then(client =>
+            client.send('Network.emulateNetworkConditions', networkConditions)
+          );
+        // Guard aganist pages that gets closed before the emulation kicks in
+        Promise.race([
+          new Promise<void>(resolve => page.on('close', () => resolve())),
+          emulatePromise,
+        ]).then(noop);
```

Comment:
NIT: I don't think we have to chain a `noop` here considering that `race` will not require any chaining to avoid throwing an unhandled rejection error.

Tested with:

```js
const p1 = () => new Promise(resolve => {
    setTimeout(() => {
        console.log("p1 resolved");
        resolve(1);
    }, 1000)
});

const p2 = () => new Promise((resolve, reject) => {
    setTimeout(() => {
        reject(2)
    }, 3000)
});

Promise.race([p1(), p2()]);

// whereas the following would throw:
// p2();
```